sdk/core/Azure.Core.TestFramework/src/RecordedTestBase.cs

```diff
@@ -195,8 +195,11 @@ protected RecordedTestBase(bool isAsync, RecordedTestMode? mode = null) : base(i
                 clientOptions.Retry.Delay = TimeSpan.FromMilliseconds(10);
                 clientOptions.Retry.Mode = RetryMode.Fixed;
             }
-
-            clientOptions.Transport = recording.CreateTransport(clientOptions.Transport);
+            // No need to set the transport if we are in Live mode
+            if (Mode != RecordedTestMode.Live)
```

Comment:
This is already handled inside of CreateTransport.

Reply:
Not exactly, it creates a ProxyTransport using the current transport, which breaks the settings needed by the client.

Reply:
It's the explicit setting of the Transport that breaks the TransportOptions, since that is interpreted by the pipeline as the customer setting the transport via clientOptions.